internal/pkg/aks/aks.go

```diff
@@ -0,0 +1,11 @@
+// Copyright 2025 Illumio, Inc. All Rights Reserved.
+
+package aks
+
+// IsAKSManagedHubble checks if the Hubble Relay service annotations indicate it's managed by AKS.
+// The annotations map is expected to come from a Kubernetes Service object, specifically
+// from the service.Annotations field.
+func IsAKSManagedHubble(annotations map[string]string) bool {
+	releaseName := annotations["meta.helm.sh/release-name"]
+	return releaseName == "aks-managed-hubble"
+}
```

Comment:
This feels very brittle, as TLS could be enabled in any Hubble Relay deployment, outside of AKS, and cloud-operator would fail in that case.

It's clear from the Cilium Helm charts that the Hubble Relay Service's port changes when TLS is enabled, from 80 to 443:
https://github.com/cilium/cilium/blob/ab227b4216c4f89cf4d2b63cc19ae5fd5e7458c0/install/kubernetes/cilium/templates/hubble-relay/service.yaml#L29-L33
Checking this port feels much more reliable.

Another, simpler method would be to simply check the existence of the `relay-client-secret.yaml` secret, as this secret is created if and only if TLS is enabled:
https://github.com/cilium/cilium/blob/ab227b4216c4f89cf4d2b63cc19ae5fd5e7458c0/install/kubernetes/cilium/templates/hubble/tls-helm/relay-client-secret.yaml#L1